api/mqs/ironmq.go

```diff
@@ -37,8 +39,7 @@ type IronMQConfig struct {
 	QueuePrefix string `mapstructure:"queue_prefix"`
 }
 
-func NewIronMQ(url *url.URL) *IronMQ {
-
+func NewIronMQ(url *url.URL, reserveTimeout time.Duration) (*IronMQ, error) {
```

Comment:
This PR is on hold in an inconsistent state while smaller pieces are merged in one at a time, starting with #599. This PR will narrow in scope and be cleaned up later, or closed in favor of another.